js/plugins/vertexai/src/index.ts

```diff
@@ -157,28 +173,35 @@ async function resolveModel(
 ) {
   const { projectId, location, vertexClientFactory } =
     await getDerivedParams(options);
-  if (actionName.includes('gemini')) {
-    const modelRef = gemini(actionName);
-    defineGeminiModel({
-      ai,
-      modelName: modelRef.name,
-      version: actionName,
-      modelInfo: modelRef.info,
-      vertexClientFactory,
-      options: {
-        projectId,
-        location,
-      },
-      debugTraces: options?.experimental_debugTraces,
-    });
-  }
-  // TODO: Support other models
+  const modelRef = gemini(actionName);
+  defineGeminiModel({
+    ai,
+    modelName: modelRef.name,
+    version: actionName,
+    modelInfo: modelRef.info,
+    vertexClientFactory,
+    options: {
+      projectId,
+      location,
+    },
+    debugTraces: options?.experimental_debugTraces,
+  });
+}
+
+async function resolveEmbedder(
+  ai: Genkit,
+  actionName: string,
+  options?: PluginOptions
+) {
+  const { projectId, location, authClient } = await getDerivedParams(options);
+
+  defineVertexAIEmbedder(ai, actionName, authClient, { projectId, location });
```

Comment:
Is the default embedder still text only? Should it be multimodal for future embedders so people can use all the capabilities of the embedders that are available?

Reply:
Good point. Made generic embedder metadata multimodal.